### ISSUE TITLE
provide a https://bun.sh/versions.json or similar version index

### ISSUE THREAD

**Issue by jcbhmr** [oven-sh/bun#6964]
### What is the problem this feature would solve?

in a scenario where you want to get the latest v0 version (but not the _latest_ latest version) you have to use the github api to list all the releases. to do that, you need to paginate since the github api only lets you go up to 100 entries at a time. and since you're making so many requests you hit the rate limit. so you need a github token. and now your cli script to `instalL-bun-v0.sh` needs a `GH_TOKEN`.

~~or another scenario: fetching all the versions of Bun in oven-sh/setup-bun to then apply a semver filter on it like denoland/setup-deno does (it supports "deno-version: v1.10.x" or similar semver ranges). if you want to fetch ALL the releases https://github.com/oven-sh/bun/releases you need to paginate and thus you need a lot of requests and thus a gh token.~~ turns out there's a special /download endpoint that provides the correct artifact (not a version list)

imo it'd be nice for these usecases to not require a GH_TOKEN github personal access token or other auth token _just to fetch all the versions_.

deno does this: https://deno.com/versions.json
node.js does this: https://nodejs.org/dist/index.json

### What is the feature you are proposing to solve the problem?

provide an api-like (but can be statically generated when a new release is created; idk) /versions.json or something to similar effect:

```jsonc
// this is an idea for format, idk if you want version numbers, tags, or what
{
  "bun": [
    // most recent at top
    "bun-v1.10.0",
    "bun-v1....",
    "bun-v1....",
    "bun-v1....",
    "bun-v0....",
    "bun-v0....",
    "bun-v0....",
    "bun-v0....",
  ]
}
```

### What alternatives have you considered?

idk. whatever works. i just want to be able to semver filter on a list of versions WITHOUT requiring any downstream users of whatever tool (cli installer locked to v0, setup-bun, whatever) to provide a GH_TOKEN

**Comment by jcbhmr** [comment]
specific enhancement relevant to bun: make it easy to get a list of all versions in https://github.com/oven-sh/setup-bun so that you can support semver ranges `bun-version: 1.0.x` or whateer and do `semver.maxSatisfying(userVersionRange, allVersions)` and then download that one

**Comment by Electroid** [comment]
> specific enhancement relevant to bun: make it easy to get a list of all versions in https://github.com/oven-sh/setup-bun so that you can support semver ranges `bun-version: 1.0.x` or whateer and do `semver.maxSatisfying(userVersionRange, allVersions)` and then download that one

You should be able to do this with `setup-bun` today, but having a version endpoint is still a good idea.

**Comment by jcbhmr** [comment]
@Electroid youre right i didn't realize that the /download endpoint supported version ranges:
this works! cool! https://bun.sh/download/1.0.x/linux/x64?avx2=true&profile=false

**Comment by jcbhmr** [comment]
how would i go about contributing website code? i can't seem to find where in the repo to do it lol
